### PR TITLE
Interactive Map Features and Database updates

### PR DESCRIPTION
**Pull request by KimberlyTruong**
**Note:** This code is not ready to be merged to main. There are still functionalities missing. This is for the sake of code review and so you all can see my progress with the project.

## Interactive Map
- The user can now specify a location (to center the map). If no location is specified, it defaults to showing a world map.
- There is code to add multiple markers to the map (right now there is only a default marker). Still need to add functionality for the user to input the locations where the markers should be.
- All map markers when generated will also render an infoWindow component. This opens when the marker is clicked on and remains on the screen until the user leaves, refreshes, or clicks the 'x' on the top right of the infoWindow.
      - This infoWindow is interactive. The user can double click to change the title of the marker at any time and allows the user to expand it to view a full note.

## Maps are no longer global
- The database is now fully developed with all the necessary tables:
      - User (userID, username, email, password)
      - Maps (mapID, mapName, center, markers, timeline)
      - Markers (infoWindow, location, note)
      - Notes (title, content)
      - Timelines (event)
- There is backend written to:
      - Create a map when the user chooses to. (integrated with front end)
      - Get all maps from the current user so they can pick which map in a dropdown menu or create a new one. (backend completed, frontend needs to be connected)
  

## Demos

ezgif com-gif-maker

ezgif com-gif-maker (1)

# Next Steps
- This pr contains all the features needed for the majority of the site. I just need to connect these.
      - Connect the search feature (where users type in addresses) to the backend to add the marker to the map.
      - Connect the dropdown menu where users select the map to display to the backend.
      - Add a feature for users to categorize their markers.
  - The above are the immediate next steps. Afterwards I will be done with maps for a while. I will begin working on grabbing directions and recommending the best mode of transportation (between walking and driving).

### CODE REVIEW

File: travel-planner/travel-planner-backend/app.js
Comment (mke135) [inline travel-planner/travel-planner-backend/app.js:4]: Nit: can we move it to a separate section to keep format looks better? 
Suggestion: move it to line 7 or 11 and add new empty lines to keep it separated from other imports.

File: travel-planner/travel-planner-backend/models/maps.js
Comment (mke135) [inline travel-planner/travel-planner-backend/models/maps.js:45]: The format(indentation) looks weird for this nested query. Is it expected?

File: travel-planner/travel-planner-backend/models/maps.js
Comment (mke135) [inline travel-planner/travel-planner-backend/models/maps.js:48]: Is this for testing? Do we want to remove it or keep it for debugging purpose when merging into master?

File: travel-planner/travel-planner-backend/routes/maps.js
Comment (mke135) [inline travel-planner/travel-planner-backend/routes/maps.js:16]: Is getAddress still broken?

**Review by mke135 — APPROVED** [review]
Great Effort! Synced up offline, the PR is getting bigger, give approval to unblock and make next review easier. Kimberly will clean up the debugging info after finish adding the feature.

**Review by davicatunda — APPROVED** [review]

File: travel-planner/travel-planner-backend/models/maps.js
Comment (KimberlyTruong) [reply travel-planner/travel-planner-backend/models/maps.js:48]: This will be removed in the next pr. This was for debugging purposes and also to see what the maps object contains to use it in other functions.

**Pull request merged**
